src/partial-templates/partial-templates.controller.ts

```diff
@@ -1,9 +1,11 @@
-import { Controller, Get, Post, Body, Patch, Param, Delete, ParseIntPipe, UseGuards } from '@nestjs/common';
+import { Controller, Get, Post, Body, Patch, Param, Delete, ParseIntPipe, UseGuards, Query, Res, HttpStatus, UseFilters } from '@nestjs/common';
 import { PartialTemplatesService } from './partial-templates.service';
 import { CreatePartialTemplateDto } from 'src/models/partialTemplate/create-partial-template.dto';
 import { UpdatePartialTemplateDto } from 'src/models/partialTemplate/update-partial-template.dto';
-import { JwtAuthGuard } from 'src/auth/strategies/guards/jwt-auth.guard';
+import { Response } from 'express';
+import { CustomExceptionFilter } from 'src/common/filters/custom-exception.filter';
 
+@UseFilters(CustomExceptionFilter)
 @Controller('partial-templates')
 export class PartialTemplatesController {
 
```

Comment:
### In your Create method, it is considered convenient to update to the following form:
```ts
@Post()
 create(@Body() createTemplateDto: CreateTemplateDto) {
 return this.templatesService.create({ ...createTemplateDto });
 }
```

### Because this approach gives us the following:
- **Clarity and simplicity:** This approach is clear and simple. It directly takes the DTO (Data Transfer Object) and passes it to the service.
- **Maintainability:** It is easy to maintain because any change to the DTO only requires a change in one place.
- **Extensibility:** If the DTO is updated, this method automatically reflects those changes without requiring additional modifications.

Reply:
Adding to what our colaboraor said, i suggest to avoid using the spread operator because of the type-safety.
According to the NestJS Documentation the controller route should look like this
```ts
@Post()
 create(@Body() createCatDto: CreateCatDto) {
   return this.catsService.create(createCatDto);
 }
```
check the documentation for more information: [Providers | NestJS - A progressive Node.js framework](https://docs.nestjs.com/providers#services)